common.yml

```diff
@@ -904,6 +912,91 @@ functions:
           fi;
           ./bin/mongo $MONGO_ARGS --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:${mongod_port}"); return true} catch(e){return false}}, "timed out connection")'
 
+  "add-aws-auth-variables-to-file":
+    - command: shell.exec
+      type: test
+      params:
+        silent: true
+        working_dir: src/github.com/mongodb/mongo-tools/common/lib
+        script: |
+          set -x
+          set -v
+          set -e
+
+          cat <<EOF > $(pwd)/aws_e2e_setup.json
+          {
+            "iam_auth_assume_aws_account": "${iam_auth_assume_aws_account}",
+            "iam_auth_assume_aws_secret_access_key": "${iam_auth_assume_aws_secret_access_key}",
+            "iam_auth_assume_role_name": "${iam_auth_assume_role_name}"
+          }
+          EOF
+
+  "setup-aws-auth-test-with-assume-role-credentials":
+    - command: shell.exec
+      type: test
+      params:
+        silent: true
```

Comment:
I'm keeping these functions as silent like in mtc since they contain private vars, but in general I don't think it matters that much.